### DISCUSSION TITLE
SD Card...

### DISCUSSION THREAD

**Discussion by bcardi0427** [mcguirepr89/BirdNET-Pi#859] [General]
I've been asking about how long an SDCard will last on BirdNETPi and Saturday I found out. My SDCard died and I'm at this point reinstalling it on a new card. I've had it running 24/7 since 1/8/2023. 4 months for the SDCard and that was a Samsung 128 gb a2 card. I've now put in a Amazon Basics 64 gb a2 card. I ordered a Libre Renegade and am going to try and see what it will do with a usb 3 Flash card or SSD hooked up and see if it can last a much longer time. I'm also looking if Clonezilla or some other program can clone the disk so I can make FULL backups of the system weekly or so.

Has anyone else had their cards go out? I'm also going to look at some of the new SBC's coming out with the Rockchip 3288 as it can use M.2's on it.

**Comment by lloydbayley** [comment]
OH YEHHHHH! Mine crapped itself.
I was doing a full image every month and a daily "just the good stuff" backup. I wrote/used a script that used dd for the image backup.

So, I replaced it with a USB flash drive and thought all was well. 
A few weeks ago, that died on me  and caused me all sorts of pain.
The bloody microSD card lasted longer than the USB flash drive for flips' sake.....

I've now put an external (real) SSD drive on it and so-far so-good.

If you want the code I was using to back up monthly via dd, I'm happy to share.
There is a SLIM possibility that it may miss a file because it runs whilst live, but I've never had a problem restoring from backup to this day. Touchwood.

**Reply by blogdemoi** [reply]
Have you heard of raspiBackup  https://github.com/framps/raspiBackup ? 

I have always wanted to try it as a Pi backup solution but never really find the time or will to try...

**Reply by lloydbayley** [reply]
Ha. You're not going to believe this but I am kinda in the same spot with raspiBackup. I saw it and thought, wow, that sounds really good, I should try it. Then I thought, well, the system I have in place is working and has proven itself but I might get around to trying it on something non-critical one day.

After spending time talking about it, I feel that one of us is now compelled to do something about trying it. Unless, we keep forgetting to... 😄

**Reply by blogdemoi** [reply]
I installed raspiBackup on another Pi (running DietPi) to test. Did my first backup. Now I have to restore the backup to see if it works, and if yes I will fine tune the backup options and go further.

**Comment by Svardsten53** [comment]
I have tried cloning the SD card to SSD storage and it worked perfectly. For about the same price as a 128GB SD card, I bought a Kingston SSD 480GB and a Deltaco mounting box for USB connection. The result is much more storage space, higher speed and significantly less risk of disk crashes.
There are tons of instructions online on how to proceed. It is a big advantage if you have an extra computer where you can backup from SD and restore to SSD. I ran this in Windows 11.
This is how I did it.


1. Image backup with Win32DiskImager (or other similar software) from SD card on another computer.
2. Create temporary SD card with updated EEPROM (Raspberry Pi model 4) plus changed boot order with RaspberryPiImager.
3. Boot the temporary card, wait until the LED flashes green quickly, turn power off.
4. Initialize the SSD in Windows with diskmgmt.msc to be able to find it when the backup is to be restored. Assign drive letter, single volume, no formatting needed, still to be overwritten later.
5. Restore image backup from step 1 to SSD with Win32DiskImager (or other similar software)
6. Remove SD card from RPi and connect SSD to blue USB port
7. Boot and check the log so everything is ok. You may have to restart services.
8. Expand storage space on SSD with "sudo raspi-config"
9. Reboot
10. Done!
